src/api/db/models/BatchError.ts

```diff
@@ -0,0 +1,73 @@
+import GraphQLError, { InternalServerError } from '../../errors.js';
+import { WRITE_IMAGES_ROLES } from '../../auth/roles.js';
+import BatchError, { BatchErrorSchema } from '../schemas/BatchError.js';
+import retry from 'async-retry';
+import { BaseAuthedModel, GenericResponse, MethodParams, roleCheck } from './utils-model.js';
+
+/**
+ * BatchErrors are errors that are generated when an uploaded Zip fails
+ * before it can be processed into it's individual images
+ * @class
+ */
+export class BatchErrorModel {
+  /**
+   * Create a new Batch Error
+   *
+   * @param {Object} input
+   * @param {String} input.batch
+   * @param {String} input.error
+   */
+  static async createError(input: { batch: string; error: string }): Promise<BatchErrorSchema> {
+    try {
+      return await retry(
+        async () => {
+          const newBatchError = new BatchError({
+            batch: input.batch,
+            error: input.error,
+          });
+          await newBatchError.save();
+          return newBatchError;
```

Comment:
Just pointing out that we're returning the entire `batchError` instead of the specific fields listed previously. I assume its ok but in case there are fields we don't want to return. However I do like the rewrite